clients/src/main/java/org/apache/kafka/common/record/FileRecords.java

```diff
@@ -421,6 +446,18 @@ private AbstractIterator<FileChannelRecordBatch> batchIterator(int start) {
         return new RecordBatchIterator<>(inputStream);
     }
 
+    /**
+     * Try populating OS page cache with file content
+     */
+    public void prepareForRead() throws IOException {
+        if (DEVNULL_PATH != null) {
+            long size = Math.min(channel.size(), end) - start;
+            try (FileChannel devnullChannel = FileChannel.open(DEVNULL_PATH, StandardOpenOption.WRITE)) {
+                channel.transferTo(start, size, devnullChannel);
```

Comment:
do we want to pre-populate the entire content represented by the FileRecords or just the one that will be read by writeTo() method? For example, we are pre-populating all content represented by FileRecords here but we might end up reading only a part of it in writeTo() (see length param in writeTo()). Can we instead stick to a pre-populating a smaller chunk, perhaps 32KB which is the chunk size used by SSL transport layer while reading from the file.

Reply:
> do we want to pre-populate the entire content represented by the FileRecords

Given that FileRecords here represents a slice of the file adjusted to max fetch bytes, we want to pre-populate the entire content here because even the single `writeTo` might read only smaller part, network-thread anyways needs entire content.